### ISSUE TITLE
Players aren't kicked on ban (Spigot)

### ISSUE THREAD

**Issue by MarioG1** [grandwazir/BanHammer#113]
Banhammer-Version: ban-hammer-3.0.0-20150201.213901-1-shaded
Spigot-Version: any (1.8)

If you ban an player he isn't kicked form the server:

Console Error: http://pastebin.com/6321Jw2G

**Comment by QueueAmI** [comment]
I've noticed this as well. Attempted to fix it myself but couldn't get it to work.

**Comment by Sandukhan** [comment]
Same problem here, testing banhammer 3.0.0 on Bukkit 1.7.2 and Spigot 1.8

1.a banned player isn't kicked from the server

2.banlimits dosent seems to work, only perm-bans possible

3.a players banhistory, thats more than 1 page long dosent show up
http://pastebin.com/d5ghF3Qr

**Comment by rustyshakelford2** [comment]
I had the same error with the kick, though timed bans are working for me. I banned myself for 1 minute and was able to get on after it expired. Also for some reason after the database upgrade /bh recent was showing bans from 2012. Bans done after the upgrade show up though.

**Comment by xLRDx** [comment]
Simple fix would be to change the commands that are all extend AbstractAsynchronousCommand to AbstractSynchronousCommand as kicking a player in a asynchronous runnable will not work needs to be synchronous

**Comment by grandwazir** [comment]
@xLRDx Simple but not ideal. The ban command does work on the database and you don't want that holding up the main thread. You could do that swap for the moment as a way to make it work but in the long term I am going to fix this by simply scheduling the creation of the event in a synchronous way.

**Comment by xLRDx** [comment]
@grandwazir yes i know it isnt ideal but needed a temp fix until you could fix the issue. I have since done sort of what you said you are going to do now but it is a bit sloppy XD. Was just trying to combat Mojang and its name changes. Been a nightmare really.

**Comment by grandwazir** [comment]
@xLRDx If you want a temporary fix I'd just get it to inherit from AbstractSynchronous it is much safer and less likely to break anything. I won't be able to commit or build anything new until Sunday I think so you might want to fix it and build it yourself in the meantime.

**Comment by xLRDx** [comment]
@grandwazir Alright thanks. Will do :+1:
